src/Symfony/Component/DependencyInjection/Compiler/ReplaceAliasByActualDefinitionPass.php

```diff
@@ -54,7 +54,13 @@ public function process(ContainerBuilder $container)
             try {
                 $definition = $container->getDefinition($targetId);
             } catch (InvalidArgumentException $e) {
-                throw new InvalidArgumentException(sprintf('Unable to replace alias "%s" with actual definition "%s".', $definitionId, $targetId), null, $e);
+                $message = sprintf('Unable to replace alias "%s" with actual definition "%s".', $definitionId, $targetId);
+
+                if ('' !== $targetId && '@' === $targetId[0]) {
+                    $message .= sprintf(' Did you mean "%s"?', substr($targetId, 1));
+                }
+
+                throw new InvalidArgumentException($message, null, $e);
```

Comment:
what about catching + throwing `ServiceNotFoundException` instead? (and not passing previous anymore?)

Reply:
I don't know, what was the reason to not throw a `ServiceNotFoundException` but `InvalidArgumentException` in the first place?

Reply:
I really don't know, especially since ServiceNotFoundException *is* an InvalidArgumentException
let's change that

Reply:
Yeah, I completely missed that `ServiceNotFoundException` extends `InvalidArgumentException`, changes done.